lib/ts/Layout.cc

```diff
@@ -140,6 +140,21 @@ Layout::check_runroot()
     runroot_map[str.substr(0, pos)] = str.substr(pos + 2);
   }
 
+  // the case for relocate (use ENV to overwrite records.config values)
+  if (runroot_map["prefix"][0] == '.') {
+    setenv("PROXY_CONFIG_CONFIG_DIR", "sysconfdir", true);
+    setenv("PROXY_CONFIG_LOCAL_STATE_DIR", "localstatedir", true);
+    setenv("PROXY_CONFIG_LOG_LOGFILE_DIR", "logdir", true);
+    setenv("PROXY_CONFIG_BIN_PATH", "bindir", true);
+    setenv("PROXY_CONFIG_PLUGIN_PLUGIN_DIR", "libexecdir", true);
+  }
+  // change it to absolute path in the map
+  for (auto it : runroot_map) {
+    if (it.second[0] == '.') {
+      runroot_map[it.first] = layout_relative(env_path, it.second.substr(2));
```

Comment:
What happens if the path is `".foo"`? Is there something that would prevent that?